Source/VCGeneration/SplitAndVerifyWorker.cs

```diff
@@ -96,50 +96,58 @@ private void TrackSplitsCost(List<Split> splits)
       }
     }
 
-    async Task DoWork(Split split, CancellationToken cancellationToken)
+    async Task DoWorkForMultipleIterations(Split split, CancellationToken cancellationToken)
+    {
+      int currentSplitNumber = DoSplitting ? Interlocked.Increment(ref splitNumber) - 1 : -1;
+      var tasks = Enumerable.Range(0, options.VcsStabilityIterations).Select(iteration =>
+        DoWork(currentSplitNumber, iteration, split, cancellationToken));
+      await Task.WhenAll(tasks);
+    }
+
+    async Task DoWork(int splitIdx, int iteration, Split split, CancellationToken cancellationToken)
```

Comment:
Can we not store the splitNumber in the Split?

Reply:
I almost removed it, but didn't because there's one place it's needed: for the `Description` property.

Now that you mention it, though, that property is never used. So I can remove it! 🎉

Reply:
Oh, never mind. It is used, thorough the `ProofRun` interface.

Reply:
Okay, but why do we pass `splitIdx` to `DoWork`, when the `Split` that we're passing also has a `splitIndex` field?